src/mapper/pkg/intentsstore/holder.go

```diff
@@ -0,0 +1,155 @@
+package intentsstore
+
+import (
+	"github.com/amit7itz/goset"
+	"github.com/otterize/network-mapper/src/mapper/pkg/graph/model"
+	"github.com/samber/lo"
+	"k8s.io/apimachinery/pkg/types"
+	"sync"
+	"time"
+)
+
+type IntentsStoreKey struct {
+	Source      types.NamespacedName
+	Destination types.NamespacedName
+	Type        model.IntentType
+}
+
+type TimestampedIntent struct {
+	Timestamp time.Time
+	Intent    model.Intent
+}
+
+type IntentsStore map[IntentsStoreKey]TimestampedIntent
+
+type IntentsHolder struct {
+	accumulatingStore IntentsStore
+	sinceLastGetStore IntentsStore
+	lock              sync.Mutex
+}
+
+func NewIntentsHolder() *IntentsHolder {
+	return &IntentsHolder{
+		accumulatingStore: make(IntentsStore),
+		sinceLastGetStore: make(IntentsStore),
+		lock:              sync.Mutex{},
+	}
+}
+
+func (i *IntentsHolder) Reset() {
+	i.lock.Lock()
+	defer i.lock.Unlock()
+
+	i.accumulatingStore = make(IntentsStore)
+}
+
+func mergeKafkaTopics(existingTopics []model.KafkaConfig, newTopics []model.KafkaConfig) []model.KafkaConfig {
+	mergedTopics := existingTopics
+	for _, newTopic := range newTopics {
```

Comment:
It may be petty, but I'm not fond of the n^2. I think it is possible doing it with 2 X lo.SliceToMap and iterating over the map or something like that.